packages/mui-material/src/Select/SelectInput.js

```diff
@@ -389,6 +389,26 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
       foundMatch = true;
     }
 
+    if (child.props.value === undefined) {
```

Comment:
@michaldudak 
This leads to crash if child is not a component but `null` or `undefined` which is a common use case when conditionals are used. First report here: https://github.com/mui/material-ui/issues/32253

What do you think?